cuqi/distribution/_cauchy.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+import scipy.stats as sps
+from cuqi.geometry import _get_identity_geometries
+from cuqi.utilities import force_ndarray
+from cuqi.distribution import Distribution
+
+class Cauchy(Distribution):
+    """ Multivariate Cauchy distribution of independent random variables.
+     
+    Each is distributed according to the PDF function
+
+    .. math::
+
+        \\frac{1}{\\pi\\gamma(1+\\frac{(x-\\mu)^2}{\\gamma^2})}
+
+    where :math:`\\mu` is the location parameter and :math:`\\gamma` is the scale parameter.
+
+    Parameters
+    ----------
+    location: float or array_like
+        Location parameter
+
+    scale: float or array_like
+        Scale parameter
+
+    Example
+    -------
+    .. code-block:: python
+
+        # % Generate a Cauchy distribution
+        import cuqi
+        location = 0
+        scale  = 1
+        x = cuqi.distribution.Cauchy(location, scale)
+        x.logpdf(0) # -1.1447298858494002
+
+    """
+
+    def __init__(self, location=None, scale=None, is_symmetric=True, **kwargs):
+        super().__init__(is_symmetric=is_symmetric, **kwargs)
+        self.location = location
+        self.scale = scale
+
+    @property
+    def location(self):
+        """ Location parameter """
+        return self._location
+    
+    @location.setter
+    def location(self, value):
+        self._location = force_ndarray(value, flatten=True)
+
+    @property
+    def scale(self):
+        """ Scale parameter """
+        return self._scale
+    
+    @scale.setter
+    def scale(self, value):
+        self._scale = force_ndarray(value, flatten=True)
+
+    def _is_out_of_bounds(self, x):
+        """ Check if x is out of bounds """
+        return np.any(self.scale <= 0)
+
+    def logpdf(self, x):
+
+        if self._is_out_of_bounds(x):
+            return -np.Inf
+        
+        return np.sum(-np.log(np.pi*self.scale*(1+((x-self.location)/self.scale)**2)))
+    
+    def cdf(self, x):
+
+        if self._is_out_of_bounds(x):
+            return -np.Inf
```

Comment:
Do we have a convention for out of bounds value? Here it is -inf, later a nan vector

Reply:
Yes convention is for logpdf, pdf and cdf to return -Inf and Nan otherwise.